js_modules/dagit/packages/core/src/gantt/GanttStatusPanel.tsx

```diff
@@ -52,10 +56,18 @@ export const GanttStatusPanel: React.FC<GanttStatusPanelProps> = ({
           break;
         case IStepState.SUCCEEDED:
           succeeded.push(key);
+          break;
       }
     }
-    return {preparing, executing, errored, succeeded};
-  }, [metadata]);
+
+    for (const node of graph) {
+      const name = node.name;
+      if (!(name in metadata.steps) || !metadata.steps[name].state) {
```

Comment:
Looks like you could probably just do:

```
if (!metadata.steps[name]?.state) {
```

Or if you want to be explicit about checking for the key, it's slightly safer to use `Object.hasOwn` instead of the `in` operator.

```
if (!Object.hasOwn(metadata.steps, 'name') || ...) {
``